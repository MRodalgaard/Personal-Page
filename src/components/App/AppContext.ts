import { createContext } from 'react';
import { AppBackground } from '../../util/constants';
import { AppColors } from '../../util/theme';

export interface IAppState {
  color: AppColors;
  background: AppBackground;
  toggleBackground?: () => void;
  toggleColor?: () => void;
}

export const initialState: IAppState = {
  background: AppBackground.greyscale,
  color: AppColors.primary,
};

const AppContext = createContext(initialState);

export default AppContext;
