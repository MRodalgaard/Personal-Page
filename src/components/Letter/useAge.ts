import { useMemo } from 'react';
import { DEFAULT_AGE } from 'util/constants';

interface IProps {
  birthday: string;
}

const useAge = ({ birthday }: IProps) => {
  return useMemo(() => {
    const today = new Date();
    const birthDate = new Date(birthday);

    if (isNaN(birthDate.getTime()) || today < birthDate) {
      return DEFAULT_AGE;
    }

    let age = today.getFullYear() - birthDate.getFullYear();
    const m = today.getMonth() - birthDate.getMonth();
    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
      age--;
    }
    return age;
  }, [birthday]);
};

export default useAge;
