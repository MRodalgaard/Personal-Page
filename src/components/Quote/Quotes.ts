export interface IQuote {
  author: string;
  text: string;
}

export class Quotes {
  private quotes: IQuote[] = [
    {
      author: 'Steve Jobs',
      text:
        "You can't just ask customers what they want and then try to give that to them. By the time you get it built, they'll want something new.",
    },
    {
      author: 'Bill Gates',
      text: 'Your most unhappy customers are your greatest source of learning.',
    },
    {
      author: 'Albert Einstein',
      text: 'Life is like riding a bicycle. To keep your balance you must keep moving.',
    },
    {
      author: 'Thomas Edison',
      text:
        'Our greatest weakness lies in giving up. The most certain way to succeed is always to try just one more time.',
    },
    {
      author: 'Pablo Picasso',
      text: 'Every child is an artist. The problem is how to remain an artist once we grow up.',
    },
    {
      author: 'Pablo Picasso',
      text: 'Computers are useless. They can only give you answers.',
    },
    {
      author: 'Leonardo da Vinci',
      text: 'Simplicity is the ultimate sophistication.',
    },
    {
      author: 'Confucius',
      text:
        'Give a man a fish and he will eat for a day. Teach a man to fish and he will eat for a lifetime.',
    },
    {
      author: 'Dr. Seuss',
      text: "Don't cry because it's over, smile because it happened.",
    },
  ];

  public getRandomQuote = () => {
    const min = 0;
    const max = this.quotes.length - 1;
    const index = Math.floor(Math.random() * (max - min + 1)) + min;
    return this.quotes[index];
  };
}
