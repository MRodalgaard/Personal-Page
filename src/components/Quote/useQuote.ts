import { useState } from 'react';

export interface IQuote {
  author: string;
  text: string;
}

const quotes: IQuote[] = [
  {
    author: 'Steve Jobs',
    text:
      "You can't just ask customers what they want and then try to give that to them. By the time you get it built, they'll want something new.",
  },
  {
    author: 'Bill Gates',
    text: 'Your most unhappy customers are your greatest source of learning.',
  },
  {
    author: 'Albert Einstein',
    text:
      'Life is like riding a bicycle. To keep your balance you must keep moving.',
  },
  {
    author: 'Thomas Edison',
    text:
      'Our greatest weakness lies in giving up. The most certain way to succeed is always to try just one more time.',
  },
  {
    author: 'Pablo Picasso',
    text:
      'Every child is an artist. The problem is how to remain an artist once we grow up.',
  },
  {
    author: 'Pablo Picasso',
    text: 'Computers are useless. They can only give you answers.',
  },
  {
    author: 'Leonardo da Vinci',
    text: 'Simplicity is the ultimate sophistication.',
  },
  {
    author: 'Confucius',
    text:
      'Give a man a fish and he will eat for a day. Teach a man to fish and he will eat for a lifetime.',
  },
  {
    author: 'Dr. Seuss',
    text: "Don't cry because it's over, smile because it happened.",
  },
  {
    author: 'C.S. Lewis',
    text: 'Two of a trade never agree.',
  },
];

interface IProps {
  initialQuote?: IQuote;
}

const useQuote = ({ initialQuote }: IProps): [IQuote, () => void] => {
  const getRandomNumber = (max: number) => {
    return Math.floor(Math.random() * (max + 1));
  };

  const getNewRandomQuote = (previousQuote?: IQuote): IQuote => {
    const newQuote = quotes[getRandomNumber(quotes.length - 1)];
    if (newQuote.text === previousQuote?.text) {
      return getNewRandomQuote(previousQuote);
    }
    return newQuote;
  };

  const [quote, setQuote] = useState(initialQuote ?? getNewRandomQuote());

  const nextQuote = () => {
    setQuote(getNewRandomQuote(quote));
  };

  return [quote, nextQuote];
};

export default useQuote;
