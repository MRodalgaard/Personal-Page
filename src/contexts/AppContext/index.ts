export { AppContextProvider } from './AppContextProvider';
export { useAppContext } from './useAppContext';
