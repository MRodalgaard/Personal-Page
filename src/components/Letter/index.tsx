import * as React from 'react';
import { useContext } from 'react';
import styled from 'styled-components';
import noteImg from '../../assets/img/note.jpg';
import analytics, { LogEvent } from '../../util/analytics';
import Age from '../Age';
import AppContext from '../App/AppContext';
import Link from '../Link';
import Quote from '../Quote';
import { IQuote } from '../Quote/Quotes';

const Container = styled.div`
  position: relative;
  background: url(${noteImg});
  border-radius: 1px;
  margin-bottom: 20px;
  padding: 20px;

  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
`;

interface IProps {
  initialQuote?: IQuote;
}

const Letter = ({ initialQuote }: IProps) => {
  const { color, toggleBackground, toggleColor } = useContext(AppContext);

  const onColorsClick = (event: React.MouseEvent<HTMLElement>) => {
    event.preventDefault();

    toggleBackground && toggleBackground();
    toggleColor && toggleColor();
    analytics.logEvent(LogEvent.ColorClick);
  };

  return (
    <Container>
      <p>
        Dear{' '}
        <Link
          href="https://whatsmybrowser.org/"
          color={color}
          onClick={() => analytics.logEvent(LogEvent.WhoAmILink)}
        >
          Reader
        </Link>
        ,
      </p>
      <p>
        <Link
          href="https://facebook.com/mrodalgaard"
          color={color}
          onClick={() => analytics.logEvent(LogEvent.FacebookLink)}
        >
          Martin Rodalgaard
        </Link>{' '}
        is a <Age birthday="08-22-1986" updateInterval={100} /> year old
        engineer living in Aarhus, Denmark, with his wonderful girlfriend Rikke
        and children Gry and Thor.
      </p>
      <p>
        He is educated in{' '}
        <Link
          href="https://linkedin.com/in/mrodalgaard"
          color={color}
          onClick={() => analytics.logEvent(LogEvent.LinkedInLink)}
        >
          Electrical Engineering
        </Link>{' '}
        specialised in Embedded Systems and{' '}
        <Link
          href="https://github.com/mrodalgaard"
          color={color}
          onClick={() => analytics.logEvent(LogEvent.GithubLink)}
        >
          Programming
        </Link>
        . He currently works at{' '}
        <Link
          href="https://trifork.com"
          color={color}
          onClick={() => analytics.logEvent(LogEvent.WorkLink)}
        >
          Trifork A/S
        </Link>{' '}
        as a Software Pilot creating mobile and web apps for customers.
      </p>
      <p>
        Martin loves{' '}
        <Link href="#" onClick={onColorsClick} color={color}>
          colors
        </Link>
        , big city traveling, skiing, watching movie classics and reading
        biographies.
      </p>
      <p>
        The gear he <u>can</u> live without (but doesn&#39;t want to) includes
        his 16" MacBook Pro, a broken iPhone X, an old Canon 550D, a converted
        Giant racing bike and an Omega Seamaster on his wrist.
      </p>
      <Quote color={color} initialQuote={initialQuote} />
    </Container>
  );
};

export default Letter;
