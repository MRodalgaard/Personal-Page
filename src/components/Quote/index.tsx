import * as React from 'react';
import { useState } from 'react';
import styled from 'styled-components';
import analytics, { LogEvent } from '../../util/analytics';
import { AppColors } from '../../util/theme';
import { IQuote, Quotes } from './Quotes';

const Container = styled.div`
  border-left: 5px solid ${AppColors.grey};
  padding-left: 10px;
  user-select: none;
  transition: color 0.5s ease;

  &:hover {
    color: ${AppColors.secondary};
    cursor: pointer;
  }
`;

const AuthorText = styled.p`
  text-align: right;
  color: ${(props) => (props.color ? props.color : AppColors.primary)};
`;

interface IProps {
  color?: string;
  initialQuote?: IQuote;
}

const Quote = ({ color, initialQuote }: IProps) => {
  const quotes = new Quotes();
  const [quote, setQuote] = useState(
    initialQuote ? initialQuote : quotes.getRandomQuote()
  );

  const onClick = () => {
    setQuote(quotes.getRandomQuote());
    analytics.logEvent(LogEvent.QuoteClick);
  };

  return (
    <Container onClick={onClick}>
      <i>{`"${quote.text}"`}</i>
      <AuthorText color={color}>- {quote.author}</AuthorText>
    </Container>
  );
};

export default Quote;
