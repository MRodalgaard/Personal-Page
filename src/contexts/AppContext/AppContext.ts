import { AppTheme } from 'contexts/ThemeContext';
import { createContext } from 'react';

export interface IAppContext {
  colorized: boolean;
  theme: AppTheme;
  toggleColorized: () => void;
  toggleTheme: () => void;
}

export const defaultValue: IAppContext = {
  colorized: false,
  theme: 'light',
  toggleColorized: () => {},
  toggleTheme: () => {},
};

export const AppContext = createContext<IAppContext>(defaultValue);
