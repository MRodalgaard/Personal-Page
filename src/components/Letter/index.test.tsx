import AppContext, { initialState } from 'components/App/AppContext';
import { IQuote } from 'components/Quote/useQuote';
import React from 'react';
import { fireEvent, render, screen } from 'util/test-utils';
import Letter from '.';

describe('Letter', () => {
  const quote: IQuote = {
    author: 'Author',
    text: 'Text',
  };

  it('renders', () => {
    const { container } = render(
      <AppContext.Provider value={initialState}>
        <Letter initialQuote={quote} />
      </AppContext.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('can click color link', () => {
    const toggleBackground = jest.fn();
    const toggleColor = jest.fn();
    const initialStateMock = { ...initialState, toggleBackground, toggleColor };

    render(
      <AppContext.Provider value={initialStateMock}>
        <Letter animate={false} initialQuote={quote} />
      </AppContext.Provider>
    );
    expect(toggleBackground.mock.calls.length).toBe(0);
    expect(toggleColor.mock.calls.length).toBe(0);

    fireEvent.click(screen.getByText('colors'));

    expect(toggleBackground.mock.calls.length).toBe(1);
    expect(toggleColor.mock.calls.length).toBe(1);
  });
});
