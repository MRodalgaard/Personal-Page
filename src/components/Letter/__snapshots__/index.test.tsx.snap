// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter renders 1`] = `
<Letter
  initialQuote={
    Object {
      "author": "Author",
      "text": "Text",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "gVpznY",
            "rules": Array [
              "
  position: relative;
  background: url(",
              "note.jpg",
              ");
  border-radius: 1px;
  margin-bottom: 20px;
  padding: 20px;

  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-ifAKCX gVpznY"
      >
        <p>
          Dear
           
          <Link
            color="#000000"
            href="https://whatsmybrowser.org/"
            onClick={[Function]}
          >
            <styled.a
              color="#000000"
              href="https://whatsmybrowser.org/"
              onClick={[Function]}
            >
              <StyledComponent
                color="#000000"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "bQBBwn",
                      "rules": Array [
                        "
  color: ",
                        [Function],
                        ";
  text-decoration: none;
  font-weight: bold;
  transition: color 0.5s ease;

  &:hover {
    color: ",
                        "#c9282d",
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.a",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                href="https://whatsmybrowser.org/"
                onClick={[Function]}
              >
                <a
                  className="sc-bwzfXH bQBBwn"
                  color="#000000"
                  href="https://whatsmybrowser.org/"
                  onClick={[Function]}
                >
                  Reader
                </a>
              </StyledComponent>
            </styled.a>
          </Link>
          ,
        </p>
        <p>
          <Link
            color="#000000"
            href="https://facebook.com/mrodalgaard"
            onClick={[Function]}
          >
            <styled.a
              color="#000000"
              href="https://facebook.com/mrodalgaard"
              onClick={[Function]}
            >
              <StyledComponent
                color="#000000"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "bQBBwn",
                      "rules": Array [
                        "
  color: ",
                        [Function],
                        ";
  text-decoration: none;
  font-weight: bold;
  transition: color 0.5s ease;

  &:hover {
    color: ",
                        "#c9282d",
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.a",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                href="https://facebook.com/mrodalgaard"
                onClick={[Function]}
              >
                <a
                  className="sc-bwzfXH bQBBwn"
                  color="#000000"
                  href="https://facebook.com/mrodalgaard"
                  onClick={[Function]}
                >
                  Martin Rodalgaard
                </a>
              </StyledComponent>
            </styled.a>
          </Link>
           
          is a 
          <Age
            birthday="08-22-1986"
            updateInterval={100}
          >
            <styled.span>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "gEHCiL",
                      "rules": Array [
                        "
  width: 20px;
  display: inline-block;
",
                      ],
                    },
                    "displayName": "styled.span",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="sc-bdVaJa gEHCiL"
                >
                  0
                </span>
              </StyledComponent>
            </styled.span>
          </Age>
           year old engineer living in Aarhus, Denmark, with his wonderful girlfriend Rikke and children Gry and Thor.
        </p>
        <p>
          He is educated in
           
          <Link
            color="#000000"
            href="https://linkedin.com/in/mrodalgaard"
            onClick={[Function]}
          >
            <styled.a
              color="#000000"
              href="https://linkedin.com/in/mrodalgaard"
              onClick={[Function]}
            >
              <StyledComponent
                color="#000000"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "bQBBwn",
                      "rules": Array [
                        "
  color: ",
                        [Function],
                        ";
  text-decoration: none;
  font-weight: bold;
  transition: color 0.5s ease;

  &:hover {
    color: ",
                        "#c9282d",
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.a",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                href="https://linkedin.com/in/mrodalgaard"
                onClick={[Function]}
              >
                <a
                  className="sc-bwzfXH bQBBwn"
                  color="#000000"
                  href="https://linkedin.com/in/mrodalgaard"
                  onClick={[Function]}
                >
                  Electrical Engineering
                </a>
              </StyledComponent>
            </styled.a>
          </Link>
           
          specialised in Embedded Systems and
           
          <Link
            color="#000000"
            href="https://github.com/mrodalgaard"
            onClick={[Function]}
          >
            <styled.a
              color="#000000"
              href="https://github.com/mrodalgaard"
              onClick={[Function]}
            >
              <StyledComponent
                color="#000000"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "bQBBwn",
                      "rules": Array [
                        "
  color: ",
                        [Function],
                        ";
  text-decoration: none;
  font-weight: bold;
  transition: color 0.5s ease;

  &:hover {
    color: ",
                        "#c9282d",
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.a",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                href="https://github.com/mrodalgaard"
                onClick={[Function]}
              >
                <a
                  className="sc-bwzfXH bQBBwn"
                  color="#000000"
                  href="https://github.com/mrodalgaard"
                  onClick={[Function]}
                >
                  Programming
                </a>
              </StyledComponent>
            </styled.a>
          </Link>
          . He currently works at
           
          <Link
            color="#000000"
            href="https://trifork.com"
            onClick={[Function]}
          >
            <styled.a
              color="#000000"
              href="https://trifork.com"
              onClick={[Function]}
            >
              <StyledComponent
                color="#000000"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "bQBBwn",
                      "rules": Array [
                        "
  color: ",
                        [Function],
                        ";
  text-decoration: none;
  font-weight: bold;
  transition: color 0.5s ease;

  &:hover {
    color: ",
                        "#c9282d",
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.a",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                href="https://trifork.com"
                onClick={[Function]}
              >
                <a
                  className="sc-bwzfXH bQBBwn"
                  color="#000000"
                  href="https://trifork.com"
                  onClick={[Function]}
                >
                  Trifork A/S
                </a>
              </StyledComponent>
            </styled.a>
          </Link>
           
          as a Software Pilot creating mobile and web apps for customers.
        </p>
        <p>
          Martin loves
           
          <Link
            color="#000000"
            href="#"
            onClick={[Function]}
          >
            <styled.a
              color="#000000"
              href="#"
              onClick={[Function]}
            >
              <StyledComponent
                color="#000000"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "bQBBwn",
                      "rules": Array [
                        "
  color: ",
                        [Function],
                        ";
  text-decoration: none;
  font-weight: bold;
  transition: color 0.5s ease;

  &:hover {
    color: ",
                        "#c9282d",
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.a",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "a",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                href="#"
                onClick={[Function]}
              >
                <a
                  className="sc-bwzfXH bQBBwn"
                  color="#000000"
                  href="#"
                  onClick={[Function]}
                >
                  colors
                </a>
              </StyledComponent>
            </styled.a>
          </Link>
          , big city traveling, skiing, watching movie classics and reading biographies.
        </p>
        <p>
          The gear he 
          <u>
            can
          </u>
           live without (but doesn't want to) includes his 16" MacBook Pro, a broken iPhone X, an old Canon 550D, a converted Giant racing bike and an Omega Seamaster on his wrist.
        </p>
        <Quote
          color="#000000"
          initialQuote={
            Object {
              "author": "Author",
              "text": "Text",
            }
          }
        >
          <styled.div
            onClick={[Function]}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "eECofK",
                    "rules": Array [
                      "
  border-left: 5px solid ",
                      "#333333",
                      ";
  padding-left: 10px;
  user-select: none;

  &:hover {
    color: ",
                      "#c9282d",
                      ";
    cursor: pointer;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
            >
              <div
                className="sc-htpNat eECofK"
                onClick={[Function]}
              >
                <i>
                  "Text"
                </i>
                <styled.p
                  color="#000000"
                >
                  <StyledComponent
                    color="#000000"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": false,
                          "lastClassName": "uGLDA",
                          "rules": Array [
                            "
  text-align: right;
  color: ",
                            [Function],
                            ";
",
                          ],
                        },
                        "displayName": "styled.p",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "p",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <p
                      className="sc-bxivhb uGLDA"
                      color="#000000"
                    >
                      - 
                      Author
                    </p>
                  </StyledComponent>
                </styled.p>
              </div>
            </StyledComponent>
          </styled.div>
        </Quote>
      </div>
    </StyledComponent>
  </styled.div>
</Letter>
`;
